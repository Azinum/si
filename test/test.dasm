0000 pushk         0
0002 assign        0
0004 pushvar       0
0006 pushk         1
0008 less_than
0009 while         10
0011 pushvar       0
0013 pushk         2
0015 add
0016 assign        0
0018 jump          -15
0020 pushk         3
0022 assign        0
0024 pushvar       0
0026 pushk         4
0028 less_than
0029 while         10
0031 pushvar       0
0033 pushk         5
0035 add
0036 assign        0
0038 jump          -15
0040 pushk         6
0042 assign        0
0044 pushvar       0
0046 pushk         7
0048 less_than
0049 while         12
0051 pushvar       0
0053 pushk         8
0055 add
0056 assign        0
0058 pushvar       0
0060 jump          -17
0062 assign        1
0064 pushk         9
0066 assign        2
0068 pushvar       2
0070 pushk         10
0072 less_than
0073 while         21
0075 pushvar       2
0077 pushk         11
0079 add
0080 assign        2
0082 pushvar       2
0084 pushk         12
0086 greater_than
0087 if            5
0089 pushk         13
0091 assign        2
0093 jump          -26
