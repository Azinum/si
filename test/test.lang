// test.lang

let i = 0;
while i < 5 {
	i = i + 1;
}

i = 0;
while (i < 5) {
	i = i + 1;
}

i = 0;
let a = while i < 5 {
	i = i + 1;
	i;
}

let q = 0;
while q < 5 {
	q = q + 1;
	if q > 3 {
		q = 12;
	}
}