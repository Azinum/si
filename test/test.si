// test.si

// Normal function
fn hello {
	let w = 10
	return w + 5
}

// Function in function
fn a {
	fn b {
		return 5
	}
	return b
}