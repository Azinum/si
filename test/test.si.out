[24 instructions, 4 variables, 0 constants]
0000 jump          11
0002 pushk         0
0004 assign        1
0006 pushvar       1
0008 pushk         1
0010 add
0011 return
0012 jump          9
0014 jump          4
0016 pushk         0
0018 return
0019 pushvar       3
0021 return
0022 pushvar       2
